lib/IMAP/MessageMapper.php

```diff
@@ -241,6 +241,34 @@ public function addFlag(Horde_Imap_Client_Socket $client,
 		);
 	}
 
+	public function getRawMessage(Horde_Imap_Client_Socket $client,
+								string $mailbox,
+								int $id): string {
+		$query = new Horde_Imap_Client_Fetch_Query();
+		$query->uid();
+		$query->fullText();
+
+
+		$result = iterator_to_array($client->fetch($mailbox, $query, [
+			'ids' => new Horde_Imap_Client_Ids($id),
+		]), false);
+
+		$msg = array_map(function (Horde_Imap_Client_Data_Fetch $result) use ($client, $mailbox) {
+			return new IMAPMessage(
+				$client,
+				$mailbox,
+				$result->getUid(),
+				$result
+			);
+		}, $result);
+
+		if (!empty($msg)) {
+			return $msg[0]->getRawMessage();
```

Comment:
so this is where `getRawMessage` is used. Can't we restructure this method to be independent of the message class? see `getHtmlBody` below.

Reply:
Indeed, thanks for your thoroughfull review.

It should also be possible, I guess, to improve this method a bit so as not to use `iterator_to_array` and `array_map` as `$id` is just a simple int. What do you think?

Reply:
`$id` is just an int, yep, but the horde method is designed to handle more than one result at a time, hence it always gives you an iterable IIRC.